### PR TITLE
Ember page embedding enhancements

### PR DESCRIPTION
**Pull request by nwmac**
This PR embed more Ember pages into the dashboard user - the goal being that net-new users do not need to switch back to the old Cluster Manager UI.

### CODE REVIEW

File: assets/translations/en-us.yaml [hunk 1]
```diff
@@ -4356,3 +4367,14 @@ support:
...
+
+v1ClusterTools:
+  monitoring:
+    label: Monitoring (Legacy)
+    description: 'Legacy V1 monitoring. V1 Monitoring is deprecated since Rancher 2.5.0. <a target="blank" href="https://rancher.com/docs/rancher/v2.x/en/monitoring-alerting/v2.5/migrating/#migrating-from-monitoring-v1-to-monitoring-v2">Learn more</a> about migrating to V2 Monitoring.'
+  logging:
+    label: Logging (Legacy)
+    description: 'Legacy V1 logging. V1 Logging is deprecated since Rancher 2.5.0. <a target="blank" href="https://rancher.com/docs/rancher/v2.x/en/logging/v2.5/migrating/">Learn more</a> about migrating to V2 Logging.'
+
+v1ProjectTools:
```
Comment (nwmac) [inline assets/translations/en-us.yaml:4379]: Remove

File: components/EmberPage.vue [hunk 2]
```diff
@@ -70,26 +82,44 @@ export default {
     ...mapGetters(['clusterId', 'productId']),
 
     loaderMode() {
-      return this.fixed ? 'content' : 'full';
+      return 'content';
```
Comment (nwmac) [inline components/EmberPage.vue:85]: Remove loader mode

File: components/EmberPage.vue [hunk 3]
```diff
@@ -184,6 +249,44 @@ export default {
...
+
+    syncHeight() {
+      if (this.heightSync) {
+        clearInterval(this.heightSync);
+      }
+
+      this.heightSync = setTimeout(() => {
+        this.doSyncHeight();
+        this.syncHeight();
+      }, 1500);
```
Comment (nwmac) [inline components/EmberPage.vue:275]: make 1s

File: components/EmberPage.vue [hunk 4]
```diff
@@ -184,6 +249,44 @@ export default {
...
+
+    doSyncHeight() {
+      if (this.inline) {
+        const iframeEl = document.getElementById(EMBER_FRAME);
+        const doc = iframeEl.contentWindow.document;
+        const app = doc.getElementById('application');
+        const h = app?.offsetHeight;
+
+        if (h && this.heightSync !== h) {
+          this.heightSync = h;
```
Comment (nwmac) [inline components/EmberPage.vue:286]: heightSync is the timer not the cached height

File: components/EmberPage.vue [hunk 5]
```diff
@@ -268,14 +394,33 @@ export default {
...
+      return false;
     }
   }
 };
 </script>
 
 <template>
-  <div class="ember-page" :class="{'fixed': fixed}">
-    <Loading :loading="!loaded" :mode="loaderMode" :no-delay="true" />
+  <div ref="emberPage" class="ember-page" :class="fixed">
```
Comment (nwmac) [inline components/EmberPage.vue:421]: Combine ember-page and fixed classes

File: components/EmberPage.vue [hunk 6]
```diff
@@ -268,14 +394,33 @@ export default {
...
+    <Loading v-if="!inline" :loading="!loaded" :mode="loaderMode" :no-delay="true" />
```
Comment (nwmac) [inline components/EmberPage.vue:422]: loaderMode is fixed as content

File: components/EmberPageView.vue [hunk 7]
```diff
@@ -28,7 +28,7 @@ export default {
 </script>
 
 <template>
-  <EmberPage v-if="src" :src="src" :fixed="true" />
+  <EmberPage v-if="src" :src="src" />
   <div v-else>
     <h1>Page Not Found</h1>
```
Comment (nwmac) [inline components/EmberPageView.vue:33]: Localize

File: components/nav/TopLevelMenu.vue [hunk 8]
```diff
@@ -228,6 +234,23 @@ export default {
               </nuxt-link>
             </div>
           </template>
+          <template v-if="legacyApps.length">
```
Comment (nwmac) [inline components/nav/TopLevelMenu.vue:237]: This should be on the if have the legacy feature flag enabled

File: config/product/legacy.js [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+  } = DSL(store, NAME);
+
+  product({
+    weight:              80,
+    ifFeature:           LEGACY,
+    showNamespaceFilter: true,
+  });
+
+  virtualType({
+    label:          'Alerts',
```
Comment (nwmac) [inline config/product/legacy.js:20]: localize

File: detail/node.vue [hunk 10]
```diff
@@ -48,13 +50,21 @@ export default {
   async fetch() {
     this.showMetrics = await allDashboardsExist(this.$store.dispatch, this.currentCluster.id, [NODE_METRICS_DETAIL_URL, NODE_METRICS_SUMMARY_URL]);
 
+    const v3Nodes = await this.$store.dispatch('rancher/request', {
```
Comment (nwmac) [inline detail/node.vue:53]: Could be gated on if v1 monitoring installed

File: detail/pod.vue [hunk 11]
```diff
@@ -98,7 +110,36 @@ export default {
...
+      if (!this.metricsID) {
+        return this.v1MonitoringUrl;
+      } else {
+        return `${ this.v1MonitoringContainerBaseUrl }/${ this.metricsID }`;
+      }
+    }
   },
+
+  methods: {
+    slectionChanged(c) {
```
Comment (nwmac) [inline detail/pod.vue:136]: typo

File: list/management.cattle.io.feature.vue [hunk 12]
```diff
@@ -89,6 +100,38 @@ export default {
         this.error = err;
         btnCB(false);
       }
+    },
+
+    doToggleWithRestart(btnCB) {
+      this.error = null;
+      try {
+        this.update.spec.value = !this.update.enabled;
+        this.update.save();
```
Comment (nwmac) [inline list/management.cattle.io.feature.vue:109]: Add comment - await can go back in when backend returns after restart

File: mixins/v1-workload-metrics.js [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+        return ns.metadata.name === namespace;
+      });
+    });
+
+    let v1MonitoringUrl = null;
+    let v1MonitoringContainerBaseUrl = null;
+
+    if (p && haveV1Monitoring(this.$store.getters)) {
+      const prjID = p.id.replace('/', ':');
+      let prefix = 'workloads/';
```
Comment (nwmac) [inline mixins/v1-workload-metrics.js:22]: Remove

File: pages/c/_cluster/explorer/index.vue [hunk 14]
```diff
@@ -151,11 +154,21 @@ export default {
 
   computed: {
     ...mapGetters(['currentCluster']),
+    ...monitoringStatus(),
 
     hideClusterToolsTip: mapPref(CLUSTER_TOOLS_TIP),
 
+    hasV1Monitoring() {
```
Comment (nwmac) [inline pages/c/_cluster/explorer/index.vue:161]: Use helper

File: pages/c/_cluster/explorer/tools/index.vue [hunk 15]
```diff
@@ -55,27 +76,68 @@ export default {
...
+      }
+
+      const chartsWithApps = charts.map((chart) => {
+        const app = this.installedAppForChart[chart.id];
+
         return {
           chart,
-          app: this.installedAppForChart[chart.id],
+          app,
+          upgradeAvailable: app?.upgradeAvailable,
```
Comment (nwmac) [inline pages/c/_cluster/explorer/tools/index.vue:98]: Change to legacy flag

File: pages/c/_cluster/explorer/tools/index.vue [hunk 16]
```diff
@@ -214,46 +337,54 @@ export default {
...
-            <template v-if="opt.app && opt.app.upgradeAvailable">
-              v{{ opt.app.currentVersion }} <b><i class="icon icon-chevron-right" /> v{{ opt.app.upgradeAvailable }}</b>
+            <template v-if="opt.app && opt.upgradeAvailable">
+              v{{ opt.app.currentVersion }} <b><i class="icon icon-chevron-right" /> v{{ opt.upgradeAvailable }}</b>
             </template>
             <template v-else-if="opt.app">
               v{{ opt.app.currentVersion }}
             </template>
-            <template v-else>
+            <template v-else-if=" opt.chart.versions.length">
```
Comment (nwmac) [inline pages/c/_cluster/explorer/tools/index.vue:355]: remove space

File: pages/c/_cluster/legacy/project/index.vue [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+        // User selected a project while on this page, so redirect to the projects view now that we have a project
+        this.$router.replace({ name: 'c-cluster-legacy-project-page', params: { page: 'apps' } });
+      }
+    }
+  }
+};
+</script>
+
+<template>
+  <IconMessage icon="icon-monitoring" message="Use the Project/Namespace filter at the top of the page to select a Project in order to see legacy Project features." />
```
Comment (nwmac) [inline pages/c/_cluster/legacy/project/index.vue:28]: Localize

File: utils/monitoring.js [hunk 18]
```diff
@@ -0,0 +1,49 @@
+// Helpers for determining if V2 or v1 Monitoring are installed
+
+import { SCHEMA, MONITORING } from '@/config/types';
+import { normalizeType } from '@/plugins/steve/normalize';
+import { findBy } from '@/utils/array';
+
+// Can be used inside a compoennts' computed property
```
Comment (nwmac) [inline utils/monitoring.js:7]: Typo

**Review by richard-cox — CHANGES_REQUESTED** [review]
- Dark mode - Dark navy background of dashboard jars with faded black background of cluster manager. As discussed I'll create a separate issue for this
- In Deployment (rancher-webhook) - Detail View - Metrics tab - the embedded window contains the warning `This is the system project which has all Kubernetes and Rancher system namespaces. Changes made to resources in the system project may harm the cluster.`. This shouldn't be there for embedded mode and we should probably show something similar in Edit View.
  - Also applies to Pod/Container metrics
  - Interestingly in edit mode we show another warning which doesn't overlap `This Deployment is managed by a Helm app; changes made here will likely be overwritten the next time the app is changed.`
- After playing around
  - i think the 1 second height ping does feel too long in places, 0.5s would be better  
  - the `Waiting for Restart` modal does need a spinner / windows progress bar 
- On a system which had v1 and v2 monitoring installed at some point (probably v1 in the same session) I disabled the `legacy` feature
  - The side nav 'LEGACY APPS / Cluster Manager' link was still shown
  - After going to the local cluster dashboard I got a white page error with the following exception 
    ```
    client.js?06a0:97 TypeError: Cannot read property 'summary' of undefined
        at _resourceCounts (ResourceSummary.vue?ef88:22) ... (`const summary = clusterCounts[resource].summary;`)
        at eval (index.vue?d30a:203)
        at Array.forEach (<anonymous>)
        at VueComponent.totalCountGaugeInput (index.vue?d30a:202)
        at Watcher.get (vue.runtime.esm.js?2b0e:4479)
        at Watcher.evaluate (vue.runtime.esm.js?2b0e:4584)
        at VueComponent.computedGetter [as totalCountGaugeInput] (vue.runtime.esm.js?2b0e:4836)
    ```
  - On refresh these two were resolved / didn't happen

**Comment by nwmac** [comment]
@richard-cox All done, apart from the dark mode change - removing the system project warning requires a change to Ember - PR for this is here: https://github.com/rancher/ui/pull/4639

**Review by richard-cox — APPROVED** [review]

**Pull request merged**